src/Playwright.MSTest/PlaywrightTimeoutAttribute.cs

```diff
@@ -0,0 +1,115 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Playwright.TestAdapter;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Playwright.MSTest;
+
+internal class PlaywrightTimeoutAttribute : TestMethodAttribute
+{
+    static internal Func<Task>? ContextCloseHookOnTimeout { get; set; } = null;
+
+    public override TestResult[] Execute(ITestMethod testMethod)
+    {
+        var timeout = PlaywrightSettingsProvider.TestTimeout;
+        if (timeout == null || Debugger.IsAttached)
```

Comment:
I get the intent of not timing out under debugger, but it makes it impossible to debug the timeout flow. It would be nice to have a ENV variable that you can set and make it go into the async flow. You will have to pump up your timeouts in the test if you want to be able to go step through it, but to me it looks useful when you are assisting someone with their issue related to timeouts.

But feel free to ignore my suggestions if you think feature creep is bigger issue than not being able to debug this.

Reply:
I'll follow-up.